### PR TITLE
Build Teleport Connect for Linux

### PR DESCRIPTION
**Pull request by zmb3**
In order to do so, we add a new make target:

    make teleterm

This (temporarily) assumes that the gravitational/webapps repo is
cloned at the right version as a sibling to the teleport repo.
(We'll be able to get rid of this when we merge webapps into Teleport)

Additionally, update dronegen to include the name of the calling
function that generated the snippet instead of the line number.
This gets rid of lot's of superfluous diffs in the generated
.drone.yml file.

Co-authored-by: Grzegorz Zdunek <grzegorz.zdunek@goteleport.com>

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by hatched — APPROVED** [review]

File: dronegen/tag.go [hunk 1]
```diff
@@ -148,6 +176,12 @@ func tagCopyArtifactCommands(b buildType) []string {
 		}
 	}
 
+	if b.hasTeleportConnect() {
+		commands = append(commands,
+			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleconnect*.{tar.gz,deb,rpm} /go/artifacts/`,
```
Comment (gzdunek) [inline dronegen/tag.go:181]: We just changed `teleconnect` to `teleport-connect` in webapps.
```suggestion
			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm} /go/artifacts/`,

**Review by gzdunek — COMMENTED** [review]
Look fine! 

I tried to run it on Drone but it failed with an error:
```
git checkout $(go run $WORKSPACE_DIR/go/src/github.com/gravitational/teleport/build.assets/tooling/cmd/get-webapps-version/main.go)
/bin/sh: go: not found
```

Also I would wait with merging this PR until https://github.com/gravitational/webapps/pull/1098 is merged.

File: dronegen/tag.go (hunk 1 above)
Comment (zmb3) [reply dronegen/tag.go:181]: Does this affect the Mac builds too?

**Comment by zmb3** [comment]
> Look fine!
> 
> I tried to run it on Drone but it failed with an error:
> 
> ```
> git checkout $(go run $WORKSPACE_DIR/go/src/github.com/gravitational/teleport/build.assets/tooling/cmd/get-webapps-version/main.go)
> /bin/sh: go: not found
> ```

I see. We don't have Go yet because this step is not running inside docker. Hm...

File: dronegen/tag.go (hunk 1 above)
Comment (gzdunek) [reply dronegen/tag.go:181]: No, this change affects only Linux packages.

**Comment by zmb3** [comment]
We're close. The build pipeline is working now, and we get an error on copying artifacts:

```
cp: can't stat '/go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm}': No such file or directory
```

File: build.assets/Dockerfile-teleterm [hunk 2]
```diff
@@ -17,9 +17,14 @@ RUN BIN="/usr/local/bin" && \
...
-ENV NODE_PATH="/usr/local/lib/node-v${NODE_VERSION}-linux-${BUILDARCH}"
+ENV NODE_PATH="/usr/local/lib/nodejs-linux"
 ENV PATH="$PATH:${NODE_PATH}/bin"
-RUN (curl -o /tmp/nodejs.tar.xz -L ${NODE_URL} && tar -xJf /tmp/nodejs.tar.xz -C /usr/local/lib)
+RUN (export NODE_ARCH=$(if [ "$BUILDARCH" = "amd64" ]; then echo "x64"; else echo "arm64"; fi) && \
+     export NODE_URL="https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-${NODE_ARCH}.tar.xz" && \
+     mkdir -p ${NODE_PATH} && \
+     curl -o /tmp/nodejs.tar.xz -L ${NODE_URL} && \
+     tar -xJf /tmp/nodejs.tar.xz -C /usr/local/lib/nodejs-linux --strip-components=1)
+RUN corepack enable yarn
```
Comment (gzdunek) [inline build.assets/Dockerfile-teleterm:28]: We need to add `rpm` tools to build `rpm`s:
```
# Install rpmbuild
RUN apt-get update -y && apt-get install -q -y --no-install-recommends rpm
```

File: build.assets/Makefile [hunk 3]
```diff
@@ -227,6 +232,24 @@ buildbox-teleterm: buildbox
 		--cache-from $(BUILDBOX_TELETERM) \
 		--tag $(BUILDBOX_TELETERM) -f Dockerfile-teleterm .
 
+
+CONNECT_VERSION ?= $(VERSION)
+ifeq ($(CONNECT_VERSION),)
+CONNECT_VERSION := $(BUILDBOX_VERSION)-dev
```
Comment (gzdunek) [inline build.assets/Makefile:238]: I'm not sure about it, for the other artifacts we have a tag name:
image

File: dronegen/tag.go [hunk 4]
```diff
@@ -148,6 +175,12 @@ func tagCopyArtifactCommands(b buildType) []string {
 		}
 	}
 
+	if b.hasTeleportConnect() {
+		commands = append(commands,
+			`cp /go/src/github.com/gravitational/webapps/packages/teleterm/build/release/teleport-connect*.{tar.gz,deb,rpm} /go/artifacts/`,
```
Comment (gzdunek) [inline dronegen/tag.go:180]: The wildcard is a problem for `cp`, we can use `find`:
```
`find /go/src/github.com/gravitational/webapps/packages/teleterm/build/release -maxdepth 1 \( -iname "teleport-connect*.tar.gz" -o -iname "teleport-connect*.rpm" -o -iname "teleport-connect*.deb" \) -print -exec cp {} /go/artifacts/ \;`
```

**Review by gzdunek — COMMENTED** [review]
As I wrote on Slack the `cp` was also failing, because the build was done for some old version of webapps (so the artifact name was `teleconnect`, not `teleport-connect`).

**Pull request merged**